engines/dgds/image.cpp

```diff
@@ -432,13 +432,13 @@ void Image::loadBitmap4(Graphics::ManagedSurface *surf, uint32 toffset, Common::
 	stream->skip(toffset >> 1);
 
 	if (highByte) {
-		for (uint32 i = 0; i < tw * th; i += 2) {
+		for (uint32 i = 0; i < static_cast<unsigned>(tw * th); i += 2) {
```

Comment:
since "unsigned" doesn't have a platform-independent size and it's compared to a uint32, it would probably be nicer to cast to `uint32` instead?

Reply:
It is confusing and a little surprising, but doing arithmetic on unsigned operands smaller than a standard word (i.e. `int`) can yield a signed integer.
Thus we get signed/unsigned mismatch warnings.

```
#include <typeinfo>
#include <cstdint>
#include <iostream>

int main() {
        uint16_t x = 45;
        uint16_t y = 834;

        auto mul = x * y;
        auto add = x + y;

        std::cout << typeid(mul).name() << std::endl;
        std::cout << typeid(add).name() << std::endl;
        std::cout << typeid(int).name() << std::endl;
        std::cout << typeid(unsigned int).name() << std::endl;
        std::cout << typeid(uint16_t).name() << std::endl;

        return 0;
}
```